contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -3229,6 +3226,54 @@ class tsqlMutator : public TSqlParserBaseListener
 	{
 		handleModuloOperator(ctx);
 	}
+	
+	void enterUnary_op_expr(TSqlParser::Unary_op_exprContext *ctx) override
+	{
+		/* 
+		 * The T-SQL grammar allows an arbitrary number of unary '+' operators to precede an expression, 
+		 * but PG only supports that for numeric expressions. For string expressions, such a '+' will raise an error in PG.
+		 * In SQL this shows as redundant operators, for example for concatenation: SELECT 'a' ++ 'b'. Expressions
+		 * such as +++(+++@v)) are also valid syntax according to the T-SQL grammar even though they look unusual.
+		 * Here we remove such unary '+' operators, which are redundant anyway. 
+		 * However we do not touch numeric constants (e.g. +123) since the '+', although still redundant, may
```

Comment:
It mentioned we do not touch numeric constants, how about the numeric variables ?

Reply:
All expressions, including numeric ones, have their unary `+` removed. If it is a unary operator, then the `+` is always redundant. This applies to variables, bracketed expressions as well as subqueries (see the test cases).
The reason we don't want to touch `+123 `is that this is a commonly known and meaningful notation by itself, which you learn in school, just like `-123`; `+@v` in contrast, is not. Arguably, there is a subjective element to this reasoning.